target/linux/ath79/dts/ar9344_engenius_ecb600.dts

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9344.dtsi"
+
+/ {
+	model = "EnGenius ECB600";
+	compatible = "engenius,ecb600", "qca,ar9344";
+
+	aliases {
+		label-mac-device = &eth0;
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		serial0 = &uart;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "ecb600:amber:power";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wmac {
+			label = "ecb600:blue:wmac";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	ath9k-leds {
+		compatible = "gpio-leds";
+
+		pcie {
+			label = "ecb600:blue:ath9k";
+			gpios = <&ath9k 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+	};
+
+	virtual_flash {
+		compatible = "mtd-concat";
+
+		devices = <&firmware1 &firmware2>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				compatible = "openwrt,okli";
+				label = "firmware";
+				reg = <0x0 0x0>;
+			};
+		};
+	};
+};
+
+&ref {
+	clock-frequency = <40000000>;
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+
+			partition@50000 {
+				label = "custom";
+				reg = <0x050000 0x050000>;
+				read-only;
+			};
+
+			partition@a0000 {
+				label = "loader";
+				reg = <0x0a0000 0x010000>;
+				read-only;
```

Comment:
Just stumbled over the read-only here. Is it really a good idea to lock the loader after it's first install? What if we want to update it at some point? Will having this rw have any downsides?

Reply:
partitioning here is exactly the same as ENS202EXT

the loader is not changed on sysupgrade.bin, only by doing another factory.bin with the failsafe image, otherwise the sysupgrade.bin would need 1.5 MB of padding.

if any bit of the loader is changed the bootloader will boot the failsafe image instead. I simply left the fakeroot partition as *not* read-only so that someone has an option to trigger failsafe mode without a serial cable.

Reply:
Not only the padding, but it would have a new MD5 checksum probably, and that would have to be updated in u-boot env

heres what u-boot env looks like

```
baudrate=115200
factory_boot=tftp 0x3000000 vmlinux-art-ramdisk && bootm 0x3000000
failsafe_boot=bootm 0x9f670000
bootcmd=bootm 0x9f0a0000
bootargs=console=ttyS0,115200 root=31:04 rootfstype=squashfs init=/etc/preinit mtdparts=ar7240-nor0:256k(u-boot),64k(u-boot-env),320k(custom),1024k(kernel),4928k(rootfs),1536k(failsafe),64k(ART) board=EAP350
bootdelay=4
ethact=eth0
ipaddr=192.168.1.1
serverip=192.168.1.101
vmlinux_start_addr=0xbf0a0000
vmlinux_size=0x000b4c67
vmlinux_checksum=d28b0788fa1395f2355ec0715d0be891
rootfs_start_addr=0xbf1a0000
rootfs_size=0x00470000
rootfs_checksum=a05bd13860789adfadd2d171c339332a
stdin=serial
stdout=serial
stderr=serial
ethaddr=00:02:6f:db:93:f6

Environment size: 685/65532 bytes
```